ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,314 @@
+        uint32_t full_input_height = output_height + dilation[0] * (kernel_size[0] - 1);
+        uint32_t full_input_width  =  output_width + dilation[1] * (kernel_size[1] - 1);
+
+        //Size of input after adding interleaved 0s.
+        uint32_t strided_input_height = (input_height - 1) * stride[0] + 1;
+        uint32_t strided_input_width  = (input_width -  1) * stride[1] + 1;
+
+        uint32_t input_pad_top  = (full_input_height - strided_input_height)/2;
+        uint32_t input_pad_bottom = full_input_height - strided_input_height - input_pad_top;
+
+        uint32_t input_pad_left = (full_input_width - strided_input_width)/2;
+        uint32_t input_pad_right = full_input_width - strided_input_width - input_pad_left;
+
+        log_debug(LogOp, "Input : {}x{}", input_height, input_width);
+        log_debug(LogOp, "Output : {}x{}", output_height, output_width);
+
+        log_debug(LogOp, "Conv Op Input : {}x{}", full_input_height, full_input_width);
+        log_debug(LogOp, "Strided Input : {}x{}", strided_input_height, strided_input_width);
+
+        log_debug(LogOp, "Padding : ({},{}) ({},{})", input_pad_top, input_pad_bottom, input_pad_left, input_pad_right);
+
+
+        DeviceComputeKernelConfig compute_kernel_config;
+        switch (device->arch()) {
+            case tt::ARCH::WORMHOLE_B0:
+                compute_kernel_config = WormholeComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity,
+                    .math_approx_mode = conv_config.math_approx_mode_enabled,
+                    .fp32_dest_acc_en = conv_config.fp32_dest_acc_enabled,
+                    .packer_l1_acc = conv_config.packer_l1_accum_enabled});
+                break;
+
+            case tt::ARCH::GRAYSKULL:
+                compute_kernel_config = GrayskullComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity, .math_approx_mode = conv_config.math_approx_mode_enabled});
+                break;
+
+            case tt::ARCH::BLACKHOLE:
+                compute_kernel_config = BlackholeComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity,
+                    .math_approx_mode = conv_config.math_approx_mode_enabled,
+                    .fp32_dest_acc_en = conv_config.fp32_dest_acc_enabled,
+                    .packer_l1_acc = conv_config.packer_l1_accum_enabled});
+                break;
+
+            default:
+                TT_THROW("Invalid Device Arch, Got {}",device->arch());
+        }
+
+
+        //Call Halo Transpose
+        auto [input_tensor_post_tm, parallel_config, tensor_manipulated, use_non_tile_height] = conv2d::shard_or_reshard_tensor_if_required(
+            device,
+            input_tensor,
+            conv_config,
+            batch_size,
+            output_height,
+            output_width,
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride,
```

Comment:
@pavlejosipovic In ConvTranspose2d, the meaning of stride & padding are the inverse of Conv2d. Auto_shard would have to check if it's the tranposed op, and if so, account for it correctly.